types/src/blob.rs

```diff
@@ -34,15 +36,18 @@ impl Blob {
     }
 
     pub fn validate(&self) -> Result<()> {
-        let computed_commitment =
-            Commitment::generate(self.namespace, self.share_version, &self.data)?;
+        let computed_commitment = Commitment::generate(self.namespace, self.to_shares()?)?;
```

Comment:
I think I prefer the following API: 

* Rename old `Commitment::generate` to `Commitment::with_blob_data` (or `Commitment::from_blob_data`?)
* Add `Commitment::with_shares` (or `Commitment::from_shares`?)

Reply:
I can just leave the `Commitment::generate` if you prefer, we don't need the version from shares

Reply:
Yes, I prefer the older API

Reply:
reverted